src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        String fruit = transaction.getFruit();
+        int amount = transaction.getAmount();
+        int currentAmount = storage.getOrDefault(fruit, 0);
+
+        if (currentAmount >= amount) {
+            storage.put(fruit, currentAmount - amount);
+        } else {
+            throw new RuntimeException("Not enough fruit in storage: " + fruit);
+        }
+    }
```

Comment:
According to SOLID principles, this method is doing more than one thing: it is validating the purchase and also updating the storage. Consider separating these responsibilities into two methods: one for validation and another for updating the storage.